core/services/job/orm.go

```diff
@@ -77,6 +78,35 @@ func NewORM(db *gorm.DB, config *storm.Config, pipelineORM pipeline.ORM, eventBr
 	}
 }
 
+func PreloadAllJobTypes(db *gorm.DB) *gorm.DB {
+	return db.Preload("FluxMonitorSpec").
+		Preload("DirectRequestSpec").
+		Preload("OffchainreportingOracleSpec").
+		Preload("KeeperSpec").
+		Preload("PipelineSpec").
+		Preload("CronSpec").
+		Preload("WebhookSpec").
+		Preload("VRFSpec")
+}
+
+func PopulateExternalInitiator(db *gorm.DB, jb Job) (Job, error) {
+	if jb.WebhookSpecID != nil {
+		// TODO: Once jpv1 is gone make an FK from external_initiators to jobs.
```

Comment:
Ticket https://app.clubhouse.io/chainlinklabs/story/10949/external-initiators-fk-to-webhook-specs